### PR TITLE
Fix `Animator` paly `BlendShape` animation bug

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

relate PR:  https://github.com/oasis-engine/engine/pull/766

### CODE REVIEW

File: packages/core/src/animation/Animator.ts [hunk 1]
```diff
@@ -610,26 +597,21 @@ export class Animator extends Component {
...
+        const srcCurve = srcCurves[srcCurveIndex].curve;
+        srcValue = this._evaluateCurve(property, srcCurve, srcClipTime, additive, srcCurve._baseOutValue);
+      } else {
+        srcValue = defaultValue;
+      }
+
+      let destValue: InterpolableValue;
+      if (destCurveIndex >= 0) {
+        const destCurve = destCurves[destCurveIndex].curve;
+        this._evaluateCurve(property, destCurve, destClipTime, additive, destCurve._crossOutValue);
```
Comment (luzhuang) [inline packages/core/src/animation/Animator.ts:611]: srcCurve._baseOutValue， destCurve._crossOutValue 不能是一个引用啊  srcCurve 可能就是 destCurve

Comment (luzhuang) [inline packages/core/src/animation/Animator.ts:611]: 是不是少了 destValue 的赋值

File: packages/core/src/animation/Animator.ts [hunk 2]
```diff
@@ -712,38 +692,34 @@ export class Animator extends Component {
     layerWeight: number,
     additive: boolean
   ): void {
-    let value: InterpolableValue;
     if (owner.type === Transform) {
       switch (owner.property) {
         case AnimationProperty.Position:
-          value = Animator._baseTempValue.vector3;
-          Vector3.lerp(srcValue as Vector3, destValue as Vector3, crossWeight, value as Vector3);
+          Vector3.lerp(srcValue as Vector3, destValue as Vector3, crossWeight, destValue as Vector3);
```
Comment (luzhuang) [inline packages/core/src/animation/Animator.ts:698]: 感觉不能直接赋值给 destValue，curve 是有可能被复用的，这里直接赋值就改变其他地方的计算结果了吧

File: packages/core/src/animation/Animator.ts (hunk 1 above)
Comment (GuoLei1990) [reply packages/core/src/animation/Animator.ts]: done

Comment (GuoLei1990) [reply packages/core/src/animation/Animator.ts]: done

**Comment by GuoLei1990** [comment]
Alreay fixed in #979

**Pull request closed without merging**